Src/Fido2/IMetadataService.cs

```diff
@@ -3,17 +3,14 @@
 
 namespace Fido2NetLib
 {
-    /// <summary>
-    /// The FIDO2 metadata service.
-    /// </summary>
     public interface IMetadataService
     {
         /// <summary>
-        /// Gets the metadata payload entry by a guid.
+        /// Gets the metadata payload entry by a guid asyncronously
         /// </summary>
         /// <param name="aaguid">The Authenticator Attestation GUID.</param>
         /// <returns>Returns the entry; Otherwise <c>null</c>.</returns>
-        MetadataBLOBPayloadEntry? GetEntry(Guid aaguid);
+        Task<MetadataBLOBPayloadEntry?> GetEntryAsync(Guid aaguid);

```

Comment:
My concern is just regarding optimization.
I have split opinions regarding this change:
I understand that keeping sync GetEntry is not really helping the library because the library code doesn't know if it the mds is optimized for sync/async. 

...But what I want to comment is:

> I don't think it is wise to remove the sync method.
> When calling the `SimpleMetdadataService` - the operation is in nature sync. Wrapping the result in a task causes overhead.
> 
> We should allow the caller to use either a sync method or async method on the interface, so *the caller* can choose what is optimal for their configuration (if they are using sync memetadataservice or async or custom)
> 
> For example, A warning or exception could be emitted from the `DistributedCacheMetadataService` if called sync.

...

But since *the caller* is the library, it has no idea to optmize for the sync mds. Any thoughts on this problem?

Reply:
SimpleMetadataService is no longer used other than by ConformanceTesting.cs so I don't think it's an issue - we could probably get rid of it entirely now.
The framework caller is already async and thus has the overhead of Task/async state machine in play already as it already awaits another Task-returning method. 

I think the actual overhead in real terms is so slight and the benefits significant enough that "if it might involve some I/O make it return Task" can be a pretty hard and fast rule in .Net server application development even if the default/mock implementation (often memory-based in libs that provide such things - IdentityServer4 for example) doesn't actually need to do anything async.

Reply:
Excellent. It should at least be properly renamed to avoid unwarranted usage.

Reply:
I'll rename/remove it later today :)